pilot/pkg/model/push_context.go

```diff
@@ -800,36 +798,44 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
+func (ps *PushContext) ServiceAttachedToGateway(hostname string, proxy *Proxy) bool {
```

Comment:
I am thinking this function call can be not very efficient

Actually, there are two loops

Reply:
IMO it's relatively trivial compared to a whole push

Reply:
What i am thinking is can we use a map

Reply:
What's the key/value of the map?

Reply:
I've removed the nested loop in `GatewayServices`, PTAL